src/Modal/Modal.tsx

```diff
@@ -138,24 +139,30 @@ export const Modal: React.FC<Props> = ({
     document.addEventListener("keydown", handleKeyDown);
 
     return () => document.removeEventListener("keydown", handleKeyDown);
-  });
+  }, [onClose]);
+
+  const type: keyof typeof motion = as.type;
+  if (!type || type === "custom") {
+    // TypeScript will give us some protection here, but we need to guarantee
+    // that `as` is an element that `motion` supports
+    throw new TypeError(
+      "`as` must be an element with a corresponding element in `Framer.motion`"
+    );
+  }
+
+  /**
+   * Framer motion component to render. The type will be taken from the `as`
+   * prop
+   */
+  const MotionComponent = motion[type];
 
   return (
     <ClassNames>
       {({ css, cx }) => {
-        const propsToPass = {
-          onClick: onClose,
-          className: classnames(
-            className,
-            cx(css(modalBackdrop)),
-            as.props.className,
-            // If the parent component is using emotion with the jsx pragma, we
-            // have to get fancy and intercept the styles to use with the
-            // `ClassNames` wrapper.
-            as.props.css ? css(as.props.css.styles) : null
-          ),
-          children: (
-            <motion.div
+        return (
```

Comment:
nbd: `{ return (` could be `(`

Reply:
Totally right. Airbnb says you should always use `=> (` instead of `=> { return (`; but I've been turned off to it over time. I often want to set up a debugger or something inside and have to change the code where I shouldn't have to.

That being said, I think we should avoid changing things to mark someone's personal preference, even if that preference is mine 😃